service/controller/resource/drainerinitializer/resource.go

```diff
@@ -209,6 +216,14 @@ func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
 				// we just stop here and move on with the other instances.
 				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no private DNS for ec2 instance %#q", *instance.InstanceId))
 				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("not draining ec2 instance %#q", *instance.InstanceId))
+
+				// Terminated instance that still have lifecycle action in the AWS API.
+				// Lets finish lifecycle hook to get rid of the instance in next loop.
+				err = r.completeLifeCycleHook(ctx, *instance.InstanceId, asgName)
```

Comment:
this will finish lifecycle hook on terminated instances ( no point doing draining as machine si already down)